local-cli/runAndroid/runAndroid.js

```diff
@@ -101,15 +102,19 @@ function buildAndRun(args, reject) {
       ? process.env.ANDROID_HOME + '/platform-tools/adb'
       : 'adb';
 
-    const adbArgs = [
-      'shell', 'am', 'start', '-n', packageName + '/.MainActivity'
-    ];
+    const devices = adb.getDevices();
 
-    console.log(chalk.bold(
-      'Starting the app (' + adbPath + ' ' + adbArgs.join(' ') + ')...'
-    ));
+    devices.forEach((device) => {
+
+      const adbArgs = ['-s', device, 'shell', 'am', 'start', '-n', packageName + '/.MainActivity'];
+
+      console.log(chalk.bold(
+        'Starting the app on ' + device + ' (' + adbPath + ' ' + adbArgs.join(' ') + ')...'
+      ));
+
+      child_process.spawnSync(adbPath, adbArgs, {stdio: 'inherit'});  
+    });
```

Comment:
Looks good. How about in case `getDevices` fails to find anything or throws for whatever reason (adb is known to be very flaky unfortunately), falling back to the original 'adb shell am start'? That way this code stays as reliable as it was but will also work in cases `adb devices` fails or returns results in an unexpected format.